### PR TITLE
add read-range-blocks admin-cmd

### PR DESCRIPTION
**Pull request by zhangchiqing**
This PR:
- Adds a `read-range-blocks` admin cmd to get blocks with a height range
- Adds a `read-range-cluster-blocks` admin cmd to get cluster blocks with a height range

### CODE REVIEW

File: admin/commands/storage/helper.go [hunk 1]
```diff
@@ -90,3 +91,33 @@ func getBlockHeader(state protocol.State, req *blocksRequest) (*flow.Header, err
...
+	endHeight, err := findUint64(input, "end-height")
+	if err != nil {
+		return nil, fmt.Errorf("invalid end-height: %w", err)
+	}
+
+	if endHeight < startHeight {
+		return nil, admin.NewInvalidAdminReqErrorf("end-height %v should not be smaller than start-height %v", endHeight, startHeight)
+	}
+
+	if endHeight-startHeight+1 > MAX_HEIGHT_RANGE {
```
Comment (pattyshack) [inline admin/commands/storage/helper.go:115]: MAX_HEIGHT_RANGE should be camel case.

File: admin/commands/storage/read_range_cluster_blocks.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+func NewReadRangeClusterBlocksCommand(db *badger.DB, headers *storage.Headers, payloads *storage.ClusterPayloads) commands.AdminCommand {
+	return &ReadRangeClusterBlocksCommand{
+		db:       db,
+		headers:  headers,
+		payloads: payloads,
+	}
+}
+
+func (c *ReadRangeClusterBlocksCommand) Handler(ctx context.Context, req *admin.CommandRequest) (interface{}, error) {
+	input, ok := req.Data.(map[string]interface{})
```
Comment (pattyshack) [inline admin/commands/storage/read_range_cluster_blocks.go:33]: parse chain id as part of parseHeightRangeRequestData?  you can pass in a requireChainID bool to the function to indicate if this should error or not

File: admin/commands/storage/read_range_blocks.go [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+}
+
+func NewReadRangeBlocksCommand(blocks storage.Blocks) commands.AdminCommand {
+	return &ReadRangeBlocksCommand{
+		blocks: blocks,
+	}
+}
+
+func (c *ReadRangeBlocksCommand) Handler(ctx context.Context, req *admin.CommandRequest) (interface{}, error) {
+	data, err := parseHeightRangeRequestData(req)
```
Comment (pattyshack) [inline admin/commands/storage/read_range_blocks.go:25]: maybe rename to reqData or reqParam.  data can refer to anything.  same elsewhere

File: admin/commands/storage/read_range_cluster_blocks.go (hunk 2 above)
Comment (zhangchiqing) [reply admin/commands/storage/read_range_cluster_blocks.go]: `parseHeightRangeRequestData` is being reused by other admin command. So I extracted a parseString to be reused

**Review by pattyshack — APPROVED** [review]
assuming you address the other comments before merging

**Review by peterargue — APPROVED** [review]

**Comment by zhangchiqing** [comment]
bors merge

**Pull request merged**